pilot/pkg/networking/core/v1alpha3/gateway_test.go

```diff
@@ -1514,6 +1514,27 @@ func TestBuildGatewayListeners(t *testing.T) {
 			},
 			[]string{"0.0.0.0_80", "0.0.0.0_8080"},
 		},
+		{
+			"gateway with bind",
+			&pilot_model.Proxy{},
+			&networking.Gateway{
+				Servers: []*networking.Server{
+					{
+						Port: &networking.Port{Name: "http", Number: 80, Protocol: "HTTP"},
+					},
+					{
+						Port:  &networking.Port{Name: "http", Number: 8080, Protocol: "HTTP"},
+						Hosts: []string{"externalgatewayclient.com"},
```

Comment:
one thing we may want to check somewhere is to have two gateways: one with localhost, and one 0.0.0.0. And make sure that virtual services can bind to a specific one. 

This example is not very practical since any virtual services would bind to both since they are in the same Gateway